File: examples/circle.jl
using Splinart

img_size = 1000
img = ones((img_size, img_size))

theta, path = circle([0.5, 0.5], 0.3, npoints = 40)

function xs_func()
    nsamples = 500
    mod2pi.(rand() .+ 2π .* LinRange(0, 1, nsamples))
end

update_img!(img, path, xs_func, theta, nrep = 1000, scale_value = 0.0001)

save_img(img, "./output", "circle.png")

show_img(img)

File: examples/circle.py
"""Example of drawing a circle with splinart."""

import numpy as np
import splinart as spl

np.random.seed(42)
img_size, channels = 1000, 4
img = np.ones((img_size, img_size, channels), dtype=np.float32)

theta, path = spl.circle([0.5, 0.5], 0.3, npoints=40)


def xs_func():
    """Return a random path on the circle."""
    nsamples = 500
    return (np.random.random() + 2 * np.pi * np.linspace(0, 1, nsamples)) % (2 * np.pi)


spl.update_img(img, path, xs_func, nrep=4000, x=theta, scale_value=0.00005)

spl.save_img(img, "./output", "circle.png")
